lib/live-collection.js

```diff
@@ -255,6 +254,18 @@ module.exports = Backbone.Collection.extend({
     }, PATCH_TIMEOUT);
   },
 
+  resubscribeOnModelChange: function (model, prop){
+    var evtName = prop ? 'change:' + prop : 'change';
+    this.listenTo(model, evtName, this.resubscribe, this);
+  },
+
+  resubscribe: function (){
+    this.reset();
+    this.subscription.cancel();
+    this.subscription = null;
+    this.listen({ snapshot: true });
```

Comment:
Additionally, it would be nice to merge the resolveChannel functionality (regarding the user) and the model into a single thing. This is probably overkill and not exactly required, but might make things simplier in the long run. 
Imagine, you had channel: `/users/:userId/rooms/:roomId` AND model `{ roomId: x, userId: y }`. Then, you just need to supply an object that had both those properties and we could make the resolveChannel functionality simpler... 

Instead of doing the stuff it does at the moment, it could simply look for all the `:param` elements and bind to changes to those elements on the model automatically. If any of them resolve to null, we don't subscribe until they're all there.

As I said, this is probably overkill, but it would make the `resolveChannel` functionality simpler...